app/assets/javascripts/covid19/src/webapp/components/structures-table/CustomCheckboxFilter.tsx

```diff
@@ -90,39 +141,75 @@ export const CustomCheckboxFilter: React.FC<CustomCheckboxFilterProps> = React.m
-    const { filterKey, filterState, setFilterState } = props;
+    const { filterKey, filterState, setFilterState, tooltip } = props;
+    const [open, setOpen] = React.useState(false);
 
-    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-        setFilterState({
-            ...filterState,
-            [event.target.name]: event.target.checked,
-        });
-    };
+    const handleChange = React.useCallback(
+        (event: React.ChangeEvent<HTMLInputElement>) => {
+            setFilterState({
+                ...filterState,
+                [event.target.name]: event.target.checked,
+            });
+        },
+        [filterState, setFilterState]
+    );
 
-    const handleClick = (event: React.MouseEvent<HTMLLIElement, MouseEvent>) => {
-        const filter = event.currentTarget.dataset.filter;
-        if (filter === undefined) return;
+    const handleClick = React.useCallback(
+        (event: React.MouseEvent<HTMLLIElement, MouseEvent>) => {
+            const filter = event.currentTarget.dataset.filter;
+            if (filter === undefined) return;
+            setFilterState({
+                ...filterState,
+                [filter]: !filterState[filter as FilterKey],
+            });
+        },
+        [filterState, setFilterState]
+    );
 
-        setFilterState({
-            ...filterState,
-            [filter]: !filterState[filter as FilterKey],
-        });
-    };
+    const closeTooltip = React.useCallback(() => {
+        setOpen(false);
+    }, []);
+
+    const openTooltip = React.useCallback(() => {
+        setOpen(true);
+    }, []);
 
     const t = React.useMemo(getTranslations, []);
+    const component = React.useMemo(
+        () => (
+            <MenuItem
+                onMouseEnter={openTooltip}
+                onMouseLeave={closeTooltip}
+                onClick={handleClick}
+                data-filter={filterKey}
+            >
+                <StyledCheckbox
+                    checked={filterState[filterKey]}
+                    onChange={handleChange}
+                    name={filterKey}
+                />
+                {t.filterKeys[filterKey]}
+            </MenuItem>
+        ),
+        [filterKey, filterState, handleChange, handleClick, t, closeTooltip, openTooltip]
+    );
 
-    return (
-        <MenuItem onClick={handleClick} data-filter={filterKey}>
-            <StyledCheckbox
-                checked={filterState[filterKey]}
-                onChange={handleChange}
-                name={filterKey}
-            />
-            {t.filterKeys[filterKey]}
-        </MenuItem>
+    return tooltip ? (
+        <HtmlTooltip title={tooltip} open={open}>
+            {component}
+        </HtmlTooltip>
+    ) : (
+        <>{component}</>
     );
 });
 
+interface FilterItemProps {
+    filterKey: FilterKey;
+    filterState: Covid19Filter;
+    setFilterState(filter: Covid19Filter): void;
+    tooltip?: NonNullable<React.ReactNode>;
```

Comment:
just out of curiosity, why the NonNullable? (as it's an optional prop)

Reply:
I applied the def type of the @material-ui tooltip.title prop. I had a mental disconnect. I was thinking of "ok, the title has this type and I have to do it optional because it will not be always present". I didn't make the relationship between optional was cancelling the NonNullable purpose. Thanks for your attention on it.